codegen/builtin_fs.go

```diff
@@ -708,6 +707,34 @@ func (dp DockerPush) Call(ctx context.Context, cln *client.Client, ret Register,
 		}
 	}
 
+	dockerAPI, err := DockerAPI(ctx)
+	if dockerAPI.Moby {
+		// Return error only if dockerPush is using docker engine instead of buildkit.
```

Comment:
Why?

Reply:
Open to another way of designing this, but basically I want hlb to function even if docker CLI is not installed. i.e. loading the docker cli could error, but if we were not using moby anyway, we don't care.